custom/icds_reports/ucr/expressions.py

```diff
@@ -314,6 +317,36 @@ class BooleanChoiceQuestion(JsonObject):
     nullable = BooleanProperty(default=True)
 
 
+icds_ucr_quickcache = get_django_quickcache(memoize_timeout=60, timeout=60 * 60)
+
+
+@icds_ucr_quickcache(('user_id',))
+def _get_user_location_id(user_id):
+    user = CommCareUser.get(user_id)
```

Comment:
nit: `CommCareUser.get_by_user_id` seems to be the preferred way to lookup users, although it probably doesn't make a difference.  That method is cached, though.

Reply:
Good point. I'm caching this quite a bit more aggressively (60 min vs 5). and now that I think about it, I shouldn't be using this method anyways as it wraps the doc which isn't strictly required.

Going to see if this helps couch for the next couple of days and then will follow up with getting this without wrapping.

Reply:
Oh that's a good point - could probably save a fair bit of processing time here if you skip the `.wrap()` call.